packages/keymaps/src/browser/user-storage-resource.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import URI from "@theia/core/lib/common/uri";
+import { Resource, ResourceResolver, Emitter, Event, DisposableCollection } from '@theia/core/lib/common';
+import { StorageServer } from '../common/user-storage-protocol';
+import { UserStorageWatcher } from '../common/user-storage-watcher';
+
+
+import { UserStorageUri } from '../common';
+
+export class UserStorageResource implements Resource {
+
+    protected readonly onDidChangeContentsEmitter = new Emitter<void>();
+    protected readonly toDispose = new DisposableCollection();
+    private _uri: string;
+
+    constructor(
+        public uri: URI,
+        protected readonly server: StorageServer,
+
+        protected readonly watcher: UserStorageWatcher
+    ) {
+
+        this._uri = uri.toString();
+        this.toDispose.push(this.onDidChangeContentsEmitter);
+        this.toDispose.push(this.watcher.onFilesChanged(changes => {
```

Comment:
Do we really need `UserStorageWatcher`?  Why not install a client on `UserStorageServer` here?

Reply:
Because I thought it overrides the last client registered, doesn't it?